app/recurringjob/job.go

```diff
@@ -0,0 +1,131 @@
+package recurringjob
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/longhorn/longhorn-manager/types"
+
+	apputil "github.com/longhorn/longhorn-manager/app/util"
+	longhornclient "github.com/longhorn/longhorn-manager/client"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+)
+
+func NewJob(name string, logger logrus.FieldLogger, managerURL string, recurringJob *longhorn.RecurringJob) (*Job, error) {
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		return nil, fmt.Errorf("failed detect pod namespace, environment variable %v is missing", types.EnvPodNamespace)
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get client config")
+	}
+	lhClient, err := lhclientset.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get clientset")
+	}
+
```

Comment:
Could we use `GetLonghornClientset` here?

NIT:
Before the func `NewJob` creates a new job, the func `recurringJob` will also use `GetLonghornClientset` to update the `RecurringJob`.
Is there any way to reuse the `lhClient` or update the `RecurringJob` in the func `NewJob`?